import_export_extensions/resources.py

```diff
@@ -247,22 +266,24 @@ def update_task_state(
     ):
         """Update state of the current event.
 
-        Receives meta of the current task and increase the `current`
-        field by 1.
+        Receives meta of the current task and increase the `current`.
 
         """
         if not current_task or current_task.request.called_directly:
             return
 
-        async_result = result.AsyncResult(current_task.request.get("id"))
-
-        self._update_current_task_state(
-            state=state,
-            meta=dict(
-                current=async_result.result.get("current", 0) + 1,
-                total=async_result.result.get("total", 0),
-            ),
-        )
+        self.current_object_number += 1
+        if (
+            self.current_object_number % self.status_update_row_count == 0
+            or self.current_object_number == self.total_objects_count
+        ):
+            self._update_current_task_state(
+                state=state,
+                meta={
+                    "current": self.current_object_number,
+                    "total": self.total_objects_count,
+                },
+            )
```

Comment:
I think this condition is complicated, so it requires more documentation: either as comment or docstring

Reply:
Suggest to try break up condition in separate vars to improve readability
```suggestion
        self.current_object_number += 1
        is_reached_update_count = self.current_object_number % self.status_update_row_count == 0
        is_last_object = self.current_object_number == self.total_objects_count
        if is_reached_update_count or is_last_object:
            self._update_current_task_state(
                state=state,
                meta={
                    "current": self.current_object_number,
                    "total": self.total_objects_count,
                },
            )
```